src/foam/u2/memento/Memento.js

```diff
@@ -411,6 +411,17 @@ foam.CLASS({
       this.memento_;
       // Copy args again cause context was needed for mementos
       this.SUPER(args);
-    }
+    },
+    function clone(opt_X) {
+      /** Create a shallow copy of this object. **/
+      var m = {};
+      for ( var key in this.instance_ ) {
+        if ( this.instance_[key] === undefined ) continue; // Skip previously cleared keys.
+
+        var value = this.instance_[key];
+        m[key] = value;
+      }
+      return this.cls_.create(m, opt_X || this.__context__);
```

Comment:
```suggestion
      return this.shallowClone(opt_X);
```
Fobjects already have a shallowClone method that does the same thing

Reply:
Or just:
% git diff ../../u2/memento/Memento.js
diff --git a/src/foam/u2/memento/Memento.js b/src/foam/u2/memento/Memento.js
index 4b5df11a2b..b711e4ea4b 100644
--- a/src/foam/u2/memento/Memento.js
+++ b/src/foam/u2/memento/Memento.js
@@ -401,6 +401,10 @@ foam.CLASS({
         return this.parentMemento_ ?
           this.Memento.create({obj: this, parent: this.parentMemento_}, this) :
           this.WindowHashMemento.create({obj: this}, this);
+      },
+      cloneProperty: function (v, m) {
+        m.memento_ = v;
       }
     }
   ],